backend/amd64/emit.mlp

```diff
@@ -1016,12 +1016,8 @@ let emit_static_cast (cast : Cmm.static_cast) i =
     I.movq (arg i 0) (res i 0)
   | Scalar_of_v128 Int32x4 -> I.movd (arg i 0) (res32 i 0)
   | V128_of_scalar Int32x4 -> I.movd (arg32 i 0) (res i 0)
-  | V128_of_scalar Float32x4 ->
-    (* CR mslater: (SIMD) remove cvt once we have unboxed float32 *)
-    I.cvtsd2ss (arg i 0) (res i 0)
-  | Scalar_of_v128 Float32x4 ->
-    (* CR mslater: (SIMD) remove cvt once we have unboxed float32 *)
-    I.cvtss2sd (arg i 0) (res i 0)
+  | V128_of_scalar Float32x4 | Scalar_of_v128 Float32x4 ->
+    if distinct then I.movss (arg i 0) (res i 0)
```

Comment:
(Arguably not part of this pull request,
but it feels like `distinct` might not be
used uniformly. Could you double
check it is always used when it can
be? For instance, shouldn't it be used
on [line 1016](https://github.com/ocaml-flambda/flambda-backend/pull/2710/files#diff-89a48d52bdba7b3a60e36cfc34c5f2f96f59da176212034bfc80e9743f9057d6R1016)?)

Reply:
I think the usage is correct - the `movq` and `movd`s are always between register classes, so are always distinct